wordpress-mobile/WordPress-Android
# Issue/12250 change site title from my site screen

Fixes #12250

This PR adds the ability to change a site title by tapping on it in the My Site screen.

I repurposed `ProfileInputDialogFragment` which had generic dialog code, suitable for like this.

You can't change site title on:
- Self Hosted sites or jetpack sites not connected to wpcom account.
- If you don't have admin privileges.

If the site title is empty site URL will be displayed instead. This is a preexisting logic.

@osullivanchris Original issue mentioned displaying a popup menu with single "Change Site Title" option, but I'm not sure if we need to add this extra step. What do you think?

Normal flow:
[![Image from Gyazo](https://i.gyazo.com/710a251f910ac8dc9bdbfb3a72a6715d.gif)](https://gyazo.com/710a251f910ac8dc9bdbfb3a72a6715d)

When the title is empty:
[![Image from Gyazo](https://i.gyazo.com/8587efc69af7cc1573d6df7a0ba5cbb3.gif)](https://gyazo.com/8587efc69af7cc1573d6df7a0ba5cbb3)

To test:
- While logged in as admin to wpcom or connected JP site tap on site title on My Site screen.
- You should see the title changer dialog.
- Updated the title and confirm that it is reflected on My Site screen and in Site Settings.
----
- Try to do the same thing without an internet connection or as a non-admin user, and confirm that you can't change the title.

One preexisting setting related issue is if you try to update settings while having a very poor internet connection and leave the app before settings are propagated to endpoint they will not be saved. 

PR submission checklist:
